### PR TITLE
Entities: Added build and buildconfig entities

### PR DESCRIPTION
**Pull request by enoodle**
Added both build and buildconfig entities and tests

### CODE REVIEW

File: lib/openshift_client.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ class Client
     # This cancels the need to define the classes
     # manually on every new entity addition,
     # and especially since currently the class body is empty
-    ENTITY_TYPES = %w(Project Route ClusterRoleBinding).map do |et|
+    ENTITY_TYPES = %w(Project Route ClusterRoleBinding Build Buildconfig).map do |et|
```
Comment (simon3z) [inline lib/openshift_client.rb]: @enoodle you should use `BuildConfigs` to respect the camel-case format.

File: test/json/build_1.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
+{
```
Comment (abonas) [inline test/json/build_1.json]: why are all the files ending with "1"? is this for "v1" ?

File: test/test_build_config.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      .to_return(body: open_test_json_file('buildconfigs_list_1.json'),
+                 status: 200)
+
+    client = OpenshiftClient::Client.new 'https://localhost:8080/oapi'
+    buildconfigs = client.get_build_configs
+
+    assert_instance_of(Kubeclient::Common::EntityList, buildconfigs)
+
+    assert_equal(1, buildconfigs.size)
+    assert_instance_of(OpenshiftClient::BuildConfig, buildconfigs[0])
```
Comment (abonas) [inline test/test_build_config.rb:20]: I suggest adding in all the tests "assert_requested"

**Comment by enoodle** [comment]
@abonas  added assert_requested and renamed the file names

File: test/test_build.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+end
+
+# Build entity tests
+class TestBuild < MiniTest::Test
+  def test_get_builds
+    stub_request(:get, %r{/builds})
+      .to_return(body: open_test_json_file('builds_list.json'),
+                 status: 200)
+
+    client = OpenshiftClient::Client.new 'https://localhost:8080/oapi'
```
Comment (abonas) [inline test/test_build.rb]: iirc, there's no need to specify /oapi when initializing the client - this is the default anyway. (it does work, but is not necessary)

**Comment by abonas** [comment]
other than the minor comment above, LGTM.
@simon3z , I'd like to see us moving away from the copy/pasting of the json files and tests. They become all too similar testing same things. VCR? Your thoughts on the topic?  (should not prevent this PR from being merged)

**Comment by enoodle** [comment]
@abonas I removed those /oapi. when only removing the oapi (leaving the slash '/' there) it didn't work as expected.

**Comment by simon3z** [comment]
> @simon3z , I'd like to see us moving away from the copy/pasting of the json files and tests. They become all too similar testing same things. VCR? Your thoughts on the topic? (should not prevent this PR from being merged)

+1 to move to VCR but we'll have to have a simple way to add/change entities without affecting the recording (re-record everything, etc.).

For example if to add a new entity we have to re-create the env and re-record the entire interactions (also for the old unrelated entities) then it's worse than the current solution.

Probably it would be best to record the interaction separately for every entity test and then we'll have to come up with something for `all_entities` (which IMO we could even remove, but that's a different discussion).

**Comment by simon3z** [comment]
@abonas @enoodle we'll probably have to add `assert_requested` to other tests as well (no need if indeed we move to VCR), other than that LGTM :+1: (ready for merge I believe).

**Comment by abonas** [comment]
@simon3z I have no objections for this to be merged.

**Pull request merged**